src/shims/unix/linux/epoll.rs

```diff
@@ -525,3 +508,28 @@ fn ready_list_next(
     }
     return None;
 }
+
+/// This helper function checks whether an epoll notification should be triggered for a specific
+/// epoll_interest and, if necessary, triggers the notification. Unlike check_and_update_readiness,
+/// this function sends a notification to only one epoll instance.
+fn check_fd_ready_event(
+    fd_ref: &FileDescriptionRef,
+    interest: Rc<RefCell<EpollEventInterest>>,
+    id: FdId,
+    ecx: &MiriInterpCx<'_>,
+) {
+    // Get the bitmask of ready events for a file description.
+    let ready_events_bitmask = fd_ref.get_epoll_ready_events().unwrap().get_event_bitmask(ecx);
```

Comment:
``get_epoll_ready_events`` will either throw_unsup_format or return an Ok(), Since we use ``unwrap()`` here this function only can be used to check file description that support epoll. Alternatively, we can let this function return an InterpResult and use ? instead of unwrap(). I am not sure which approach is better.

Reply:
Oh right, we don't have a test for inserting an FD (that doesn't support epoll) into an epoll FD. Please add such a test. Pretty sure it should hit that unwrap

Reply:
Yup, it hits the unwrap. ``InterpResult`` is better.